src/finite_difference.c

```diff
@@ -0,0 +1,66 @@
+#include <finite_difference.h>
```

Comment:
I think this kind of import might cause errors once you try to build the prototype